### PR TITLE
Fix 1D plots to properly adjust zoom level when changing traces.

### PR DESCRIPTION
**Pull request by adamcw**
Fixes issue #242 where 1D plots do not re-zoom to properly frame the data when traces are toggled on and off.

Includes an update to `data_demo.py` to create sine waves with enough amplitude variance to test this feature in the future to avoid further regressions.

### CODE REVIEW

File: client-js/app/elements/labrad-plot.ts [hunk 1]
```diff
@@ -367,8 +367,18 @@ export class Plot extends polymer.Base {
 
     this.lastData = null;
 
+    this.dataLimits.xMin = 0;
+    this.dataLimits.xMax = 0;
+    this.dataLimits.yMin = 0;
+    this.dataLimits.yMax = 0;
```
Comment (maffoo) [inline client-js/app/elements/labrad-plot.ts]: Should use `NaN` as the sentinel values when data limits need to be recomputed. Might be good to modify the test case to put some traces entirely above or below 0 to check this. Also, probably want to reset `zMin` and `zMax` to take care of the 2D case, and add a data_demo case for that too.

Comment (adamcw) [inline client-js/app/elements/labrad-plot.ts]: Existing 2D_multi data test is sufficient to test the 2D case. It elucidated the fact the z-axis wasn't being adjusted correctly when all values were equal, so I fixed that also.

**Comment by adamcw** [comment]
PTAL.

File: server/src/test/scripts/data_demo.py [hunk 2]
```diff
@@ -36,7 +36,6 @@ def demo_1d_simple(dv):
         dv.add([x, y])
         time.sleep(0.002)
 
-
```
Comment (maffoo) [inline server/src/test/scripts/data_demo.py:39]: style guide says: two blank lines between top-level definitions (https://google.github.io/styleguide/pyguide.html#Blank_Lines)

**Comment by maffoo** [comment]
One minor comment about whitespace. Otherwise LGTM.

**Pull request merged**